pystac_client/client.py

```diff
@@ -497,6 +499,29 @@ def get_search_link(self) -> Optional[pystac.Link]:
             None,
         )
 
+    def get_queryables(self) -> Dict[str, Any]:
+        """Return all queryables.
+
+        Output is a dictionary that can be used in ``jsonshema.validate``
+
+        Return:
+            Dict[str, Any]: Dictionary containing queryable fields
+        """
+        assert self._stac_io is not None
+        self._stac_io.assert_conforms_to(ConformanceClasses.FILTER)
+
+        self_href = self.get_self_href()
+        if self_href is None:
+            raise ValueError("cannot build a queryable href without a self href")
+
+        url = f"{self_href.rstrip('/')}/queryables"
```

Comment:
So this is where the trickiness about configuration (e.g. https://github.com/stac-utils/pystac-client/issues/320#issuecomment-1255202184) comes into play. Technically, the landing page should have a `queryables` link: https://github.com/stac-api-extensions/filter#interaction-with-endpoints. But not many do (e.g. the Planetary Computer doesn't). 

I think the current implementation is fine for this PR, but eventually we'll want to refactor link building out to a single spot (probably a method on `Client` that can be configured w/ attributes on the client).

Just a comment, no changes required on this line.

Reply:
:+1: Yeah I saw that. I can try to make it clearer that this will not work forever.